### PR TITLE
Reconfigurations and chores

### PR DESCRIPTION
**Pull request by xeho91**
Please see each commits for explanations. I did my best to keep the commits ordered and descriptive enough. :smile:

Let me know if you catch any bugs/problems.

**What possibly could be done next?**
1. Discuss if we could/should add auto-fixing via linters (if possible) and do it while staging _(with `git add ...` command)_
2. Research for adding GitHub automation(actions?) for Pull Requests

### CODE REVIEW

**Comment by mariushilleke** [comment]
> 1. Discuss if we could/should add auto-fixing via linters (if possible) and do it while staging _(with `git add ...` command)_
For the `git add` command it sounds reasonable, since it is a good step in the version control process to add automatic adjustments, since it is not the final commit but "only" staging the changes that are going to be committed, but we can maybe open a thread for this discussion with a poll
> 2. Research for adding GitHub automation(actions?) for Pull Requests
Probably @lax0248917 has already a lot of levers to pull on for that topic.

**Comment by mariushilleke** [comment]
just feedback for the commit-history
👍 clean and very helpful

File: package.json [hunk 1]
```diff
@@ -1,66 +1,121 @@
...
+		"eslint": "^7.10.0",
+		"eslint-config-prettier": "^6.12.0",
 		"eslint-import-resolver-alias": "^1.1.2",
-		"eslint-plugin-import": "^2.22.0",
-		"husky": "^4.2.5",
-		"prettier": "^2.0.5",
-		"webpack": "^4.43.0",
+		"eslint-plugin-import": "^2.22.1",
+		"eslint-plugin-node": "^11.1.0",
+		"husky": "^4.3.0",
```
Comment (mariushilleke) [inline package.json:99]: Shall we use `husky v5`? It is free to use for open source projects?

Comment (mariushilleke) [reply package.json:99]: But still alpha-stage, maybe waiting is better.

Comment (xeho91) [reply package.json:99]: Is still in alpha, so until it becomes stable, we can migrate, as if there are any bugs, then only Sponsors get support.

File: package.json [hunk 2]
```diff
@@ -1,66 +1,121 @@
...
+		"check:all": "pnpm check \".\"",
+		"check:js": "pnpm check \"./**/*.js\"",
+		"check:json": "pnpm check \"./**/*.json\"",
+		"check:md": "pnpm check \"./**/*.md\"",
+		"commit": "pnpm lint:staged && git cz --no-verify",
+		"lint:all": "pnpm lint:js:all && pnpm lint:md:all && pnpm lint:spell:all",
+		"lint:git": "commitlint --config \"./config/.commitlintrc.js\" --verbose",
+		"lint:git:all": "pnpm lint:git -- --from \"develop\" --to \"HEAD\"",
+		"lint:git:last": "pnpm lint:git -- --from HEAD~1",
+		"lint:js": "eslint --config \"config/.eslintrc.js\" --ignore-path \"./config/.eslintignore\" --ext \"js\"",
```
Comment (mariushilleke) [inline package.json:52]: So far there does not seem to be a `verbose` option for eslint, right?

File: package.json [hunk 3]
```diff
@@ -1,66 +1,121 @@
...
 		"type": "git",
 		"url": "https://github.com/TeamArmadilloOrg/lilly-cli.git"
 	},
-	"bugs": "https://github.com/TeamArmadilloOrg/lilly-cli/issues",
-	"engines": {
-		"node": "^12.18.3",
-		"pnpm": "^5.5.0",
-		"vscode": "^1.48.0"
-	},
 	"scripts": {
```
Comment (mariushilleke) [inline package.json:38]: Is it possible to use single quotes, as well, instead of escaping double quotes for all the paths?

File: config/webpack.config.js [hunk 4]
```diff
@@ -84,8 +83,13 @@ module.exports = {
 	plugins: [
 		new CleanWebpackPlugin({
 			// https://github.com/johnagan/clean-webpack-plugin#options-and-defaults-optional
+			verbose: true,
 		}),
 
-		new webpack.BannerPlugin({ banner: "#!/usr/bin/env node", raw: true }),
+		new webpack.BannerPlugin({
```
Comment (mariushilleke) [inline config/webpack.config.js:89]: Very cool plugin. Did you add the shebang manually, before?

File: package.json (hunk 2 above)
Comment (xeho91) [reply package.json:52]: There is no such option (flag) available in ESLint CLI. You can verify it by running for example: `pnpx eslint --help`.

The closest is `--debug`. You can try passing it with `pnpm lint:js:all -- -- --debug`, but the output is crazy 😅 

**FYI**: why so many  double dashes `--`?
Because of:
https://stackoverflow.com/questions/43046885/what-does-do-when-running-an-npm-command
tl;dr:
We want to pass `--debug` flag to the ESLint itself.
So, we're asking to not pass it to `pnpm` and not to `pnpm` again. Look, an inception concept, eh?

File: package.json (hunk 3 above)
Comment (xeho91) [reply package.json:38]: Sorry, I'm confused. What paths?

JSON format doesn't accept single quotes, as far I know.

File: config/webpack.config.js (hunk 4 above)
Comment (xeho91) [reply config/webpack.config.js:89]: Nothing changed here, I just changed the style format a bit, so I could add the comments.

File: package.json (hunk 3 above)
Comment (xeho91) [reply package.json:38]: If you're talking about paths to the config for the tools, then I shall remind you that you had the problem with them on Windows. I verified it myself using PowerShell.

File: config/.babelrc.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
-var babelConfig = {
```
Comment (mariushilleke) [inline config/.babelrc.js:1]: Did you use `.babelrc.js` instead of `babel.config.json` because the later one should be placed in the root directory to work properly and therefore it was not an option in the existing `root/config` directory? This solution seems more explicit about which directories should be used for bablelizing, as well.

Root of this question >>> the ["node" script in `package.json`](https://github.com/TeamArmadilloOrg/lilly-cli/pull/16/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R59) which aliases the normal node-command with the babled version and this respective config.

Comment (mariushilleke) [reply config/.babelrc.js:1]: PS: if you click the link to package.json:59, it will bring you there directly, but unfortunately the 59th row is hidden right at the top of the browsers scroll-windows...
At least it was in my case like this, when I tested the link.

Comment (xeho91) [reply config/.babelrc.js:1]: > Did you use .babelrc.js instead of babel.config.json because the later one should be placed in the root directory to work properly and therefore it was not an option in the existing root/config directory? This solution seems more explicit about which directories should be used for bablelizing, as well.

Well, no.
Following this: https://babeljs.io/docs/en/config-files#supported-file-extensions

I prefer `*.js` config file over `*.json`, because:
- I can write comments
- I can make it look prettier, with new lines, etc. I am not limited to JSON format
- I can write some advanced functions, when they're needed
- **Types**! I know there's such thing as JSON thema, but types work better for me in my honest opinion
- I don't need to worry about caching, as for now, because my project(s) doesn't have such problems

The path to the config file is not a problem, thanks to this option:
https://babeljs.io/docs/en/babel-cli#custom-config-path
So, it doesn't matter if it was located in the root or somewhere else. They're not highlighting it well enough, as there seems to be a standard convention in the projects to put every configuration files in the root directory - which in my opinion is making this horribly messy and overwhelming, especially for the beginners.

Comment (xeho91) [reply config/.babelrc.js:1]: > PS: if you click the link above ( "node" script in package.json) to package.json:59, it will bring you there directly, but unfortunately the 59th row is hidden right at the top of the browsers scroll-windows...
At least it was in my case like this, when I tested the link.

Yeah, I noticed. Because of the GitHub header being "sticky", and blocking the view of element id being "scrolled to". 😄 This is a very common mistake made in web development. You should report it :+1:

Comment (xeho91) [reply config/.babelrc.js:1]: > Root of this question >>> the "node" script in package.json which aliases the normal node-command with the babled version and this respective config.

Technically, this is not an alias. You still have to call it with `pnpm node`, which instead of using standard `node`, it will use `babel-node` _(and with specified configuration loaded in advanced, thanks to the option passed `--config \"./config/.babelrc.js\"`)_.

If you wont pass any file, you can run `babel-node-REPL`, but this isn't really useful, as not everything works here. Is better to run/compile the file instead.
Source: https://babeljs.io/docs/en/babel-node#es6-style-module-loading-may-not-function-as-expected

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,31 +1,85 @@
...
+   information which hasn't been included in the issue or the commit messages.
+6. Wait for the approvals from the core team.
+
+[new-issues]: https://github.com/TeamArmadilloOrg/lilly-cli/issues/new/choose
+[conventional-commits]: https://conventionalcommits.org
+
+## Providing feedback
+
+Your comments and feedback are welcome, and the development team is going to be
+available to you via a handful of different channels.
```
Comment (mariushilleke) [inline CONTRIBUTING.md:65]: Should we already mention channels here?

Comment (xeho91) [reply CONTRIBUTING.md:65]: As soon as we're ready, yes. This is still a draft.

File: package.json (hunk 2 above)
Comment (mariushilleke) [reply package.json:52]: Because using the script from the `package.json` would result in
`pnpm lint:js:all` 
which would resolve to
`pnpm pnpm lint:js .`
- actually why is a second pnpm in the scripts-value for key `lint:js:all` necessary?
which would resolve to 
`pnpm pnpm eslint --config config/.eslintrc.js --ignore-path config/.eslintignore --ext js`
- for consistency reasons the second dot-reference for the `--ignore-path` in line 52 option could be deleted, right?

So adding the kind of verbose-option `--debug` to that script it has to be handed over to eslint itself, as you mentioned, and therefore the `--` double-dashes tell the CLI to skip the first tool `pnpm`, the second tool as well `pnpm` and then applying as option to the last tool which then is `eslint`, ufff :D man no let's skip this option on this one. Is hard to read and follow.

File: package.json (hunk 3 above)
Comment (mariushilleke) [reply package.json:38]: Although you are completely

> Sorry, I'm confused. What paths?

For e.g. all the files handed over to the `--config`-option.

> JSON format doesn't accept single quotes, as far I know.

Even wrapped in the double quotes?

Comment (xeho91) [reply package.json:38]: Well, if we wrap the single quotes with double quotes, we would return to the previous problem you had:
https://github.com/TeamArmadilloOrg/lilly-cli/pull/15#discussion_r488385764

File: .vscode/cspell.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+		"bash",
+		"!aws"
+	],
+	"allowCompoundWords": false,
+	"words": [
+		"CODEOWNERS",
+		"mariushilleke",
+		"pnpm",
+		"rebased",
+		"xeho"
```
Comment (lax0248917) [inline .vscode/cspell.json:27]: Are these words that will no longer appear as spelling errors? Can we add lax0248917 to this list? ;)

File: .vscode/extensions.jsonc [hunk 8]
```diff
@@ -0,0 +1,49 @@
+{
+	// See https://go.microsoft.com/fwlink/?LinkId=827846 to learn about
+	// workspace recommendations.
+	// Extension identifier format: ${publisher}.${name}. Example: vscode.csharp
+	// List of extensions which should be recommended for users of this
+	// workspace.
+	"recommendations": [
```
Comment (lax0248917) [inline .vscode/extensions.jsonc:7]: Have we established a preferred OS? Do this extension reflect on that OS? Or can a Mac user expect as much love and support as well 😆

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,31 +1,85 @@
...
 
-Thank you very much for sparking an interest in contributing to this our side project. 😊\
-Keep in mind that this project is released with a [Contributor Code of Conduct](CODE_OF_CONDUCT.md).
-By contributing to this project you agree to abide its terms.
-
-If you have any questions which haven't been addressed here, in our Wiki or in the
-[README.md](README.md),\
-please feel free to create a issue or let us know (@xeho91 🌅,
-@mariushilleke 🐢, @lax0248917 🚀) on our Slack channel.
+Thank you very much for sparking an interest in contributing to this our side
```
Comment (lax0248917) [inline CONTRIBUTING.md:3]: Suggest editing this to read as "Thank you very much for sparking an interest in contributing to our side project."

**Review by lax0248917 — APPROVED** [review]
Apologize for the delay in review. Awesome job guys and am really impressed with all the work that has been put into this project. You guys are awesome!!!

File: .vscode/cspell.json (hunk 7 above)
Comment (xeho91) [reply .vscode/cspell.json]: Correct, added.

File: CONTRIBUTING.md (hunk 9 above)
Comment (xeho91) [reply CONTRIBUTING.md]: Fixed

File: .vscode/extensions.jsonc (hunk 8 above)
Comment (xeho91) [reply .vscode/extensions.jsonc]: Of course. I had no way to verify if these extensions work in MacOS, but as far I am concerned, there should be no issues.

I would rather expect more problems with Windows though 😝

File: package.json (hunk 2 above)
Comment (xeho91) [reply package.json:52]: >**which would resolve to**
>
> `pnpm pnpm lint:js .`

No, technically it would resolve to `pnpm lint:js "."` 😜


What if we wanted to lint just specific file? For example for the **lint-staged** tool?
Hence why I created two scripts.
If we decide to change some properties, such as changing config locations, adding new options, etc, we would have to change those two lines, instead of just one. Hence why I made it more dynamic. It saves us from comparing/writing those two lines like:
```json
"scripts:" {
  "pnpm lint:js": "eslint --config config/.eslintrc.js --ignore-path config/.eslintignore --ext js",
  "pnpm:lint:all": "eslint --config config/.eslintrc.js --ignore-path config/.eslintignore --ext js ."
```

Also, `--debug` option from ESLint is completely useless for us. Unless we have some problem with this tool or config.

Comment (xeho91) [reply package.json:52]: On the second though, I have no problem to remove `pnpm:lint:all`, as we could just use `pnpm lint:js .`.

Yay or nay?

File: package.json (hunk 3 above)
Comment (mariushilleke) [reply package.json:38]: > Well, if we wrap the single quotes with double quotes, we would return to the previous problem you had:
> [#15 (comment)](https://github.com/TeamArmadilloOrg/lilly-cli/pull/15#discussion_r488385764)

Sorry, for letting you searching for the last issue concerning that topic. I can see your point, though then it probably is another issue about marking the paths with quotes (let it be single or double) or leaving the quotes at all, where you mentioned in this [#15 (comment)](https://github.com/TeamArmadilloOrg/lilly-cli/pull/15#discussion_r488385764):

>Removing quotes is a bad idea (less readable, might be confusing in the future and create new problems).

```
"lint:js": "eslint --config \"./config/.eslintrc.js\" --ext js",
```
or

```
"lint:js": "eslint --config ./config/.eslintrc.js --ext js",
```
Actually I have to confess that I think the second solution is more readable without the need for escaping additional double-quotes by back-slashes, since it is still clear that it is an argument handed over to the option `config`, don't you think? Which other problems, that will be created by leaving the double-quotes, were you talking about?

Comment (xeho91) [reply package.json:38]: https://medium.com/@jakubsynowiec/you-should-always-quote-your-globs-in-npm-scripts-621887a2a784

Note: In this article, he's not mentioning the issue with single quotes in with Git Bash For Windows.
It was mentioned in the comment:
https://medium.com/@pkursawe/be-aware-that-on-the-windows-git-bash-you-have-to-use-double-quotes-instead-of-single-ones-in-your-45dbb7fa9145
